chapter_recurrent-neural-networks/encoder-decoder.md

```diff
@@ -1,20 +1,19 @@
 # Encoder-Decoder Architecture
 
-The encoder-decoder architecture is a neural network design pattern. In this architecture, the network is partitioned into two parts, the encoder and the decoder. The encoder's role is encoding the inputs into state, which often contains several tensors. Then the state is passed into the decoder to generate the outputs. In machine translation, the encoder transforms a source sentence, e.g., "Hello world.", into state, e.g., a vector, that captures its semantic information. The decoder then uses this state to generate the translated target sentence, e.g., "Bonjour le monde.".
+The *encoder-decoder architecture* is a neural network design pattern. In this architecture, the network is partitioned into two parts, the encoder and the decoder. The encoder's role is to encode the inputs into state, which often contains several tensors. Then the state is passed into the decoder to generate the outputs. In machine translation, the encoder transforms a source sentence, e.g., "Hello world.", into state, e.g., a vector, that captures its semantic information. The decoder then uses this state to generate the translated target sentence, e.g., "Bonjour le monde.".
 
 ![The encoder-decoder architecture.](../img/encoder-decoder.svg)
 
 In this section, we will show an interface to implement this encoder-decoder architecture.
 
-```{.python .input  n=1}
-from mxnet.gluon import nn
-```
 
 ## Encoder
 
-The encoder is a normal neural network that takes inputs, e.g., a source sentence, to return outputs.
+The encoder is a normal neural network that takes inputs, e.g., a source sentence, and then returns states, which is a feature map in a form of vector or tensor.
```

Comment:
"states" is not a singular, why using "which is"?

Reply:
noticed